File: pset6/vigenere.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("Usage python vigenere.py key")
        exit(1)

    k=sys.argv[1].upper()

    for i in k:
        if not i.isalpha():
            print("Alphabetic key only")
            exit(2)

    print("plaintext: ", end="")
    plaintext=cs50.get_string()
    if plaintext!= None:
        ciphertext=encrypt(plaintext, k)
        print("ciphertext= {}".format(ciphertext))

def encrypt(text, key):
    counter=0
    ciphertext=""
    for i in text:
        k=ord(key[counter])-65
        if i.isalpha() and i.isupper():
            c= ord(i)
            c= (c-65+k)%26;
            c+=65;
            ciphertext += chr(c)
            counter = (counter+1)%len(key)

        elif i.isalpha() and i.islower():
            c= ord(i)
            c= (c-97+k)%26;
            c+=97;
            i= chr(c)
            ciphertext += chr(c)
            counter = (counter+1)%len(key)

        else:
            ciphertext += i
    return ciphertext

if __name__ == "__main__":
    main()

File: pset2/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int i, key_counter;
string key, plaintext;

void encrypt(string s)
{
        for (i=0; i<strlen(s); i++)
    {
        if (isalpha(s[i]) && isupper(s[i]))
        {
          
            int c= (int) s[i];
            c= (c-65+(int) key[key_counter]-65)%26;
            c+=65;
            s[i]= (char) c;
            key_counter=(key_counter+1)%strlen(key);
        }
        else if (isalpha(s[i]) && islower(s[i]))
        {
            int c= (int) s[i];
            c= (c-97+((int) key[key_counter])-65)%26;
            c+=97;
            s[i]= (char) c;
            key_counter=(key_counter+1)%strlen(key);
            
        }
    }
}
int main(int argc, string argv[])
{
    if (argc==2)
        {
        key= argv[1];
        for (i=0; i<strlen(key); i++)
            {
            if (isalpha(key[i]))
            {
                key[i]=toupper(key[i]);
            }
            else
            {
                printf("Alphabetic key only\n");
                return 1;
            }
            }
    printf("plaintext: ");
    plaintext=get_string();
        if (plaintext!=NULL)
        {
            key_counter=0;
            encrypt(plaintext);
            printf("ciphertext: %s\n", plaintext);
        }
        }
   
    else
    {
        printf("Usage: ./vigenere k\n");
        return 1;
    }
}